src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -99,16 +100,13 @@ public void hitExecute(SearchContext context, HitContext hitContext) {
 
                 String highlighterType = field.fieldOptions().highlighterType();
                 if (highlighterType == null) {
-                    boolean useFastVectorHighlighter = fieldMapper.fieldType().storeTermVectors() && fieldMapper.fieldType().storeTermVectorOffsets() && fieldMapper.fieldType().storeTermVectorPositions();
-                    if (useFastVectorHighlighter) {
-                        highlighterType = "fvh";
-                    } else if (fieldMapper.fieldType().indexOptions() == IndexOptions.DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS) {
-                        highlighterType = "postings";
-                    } else {
-                        highlighterType = "plain";
+                    for(String highlighterCandidate : sortedHighlighters) {
+                        if (highlighters.get(highlighterCandidate).canHighlight(fieldMapper)) {
+                            highlighterType = highlighterCandidate;
+                            break;
+                        }
                     }
```

Comment:
can we add an assert to make sure that highlighterType != null here? it really should since we know that plain highlighter always returns true, but the assert would make it more explicit that it is expected